nested-folders/src/Folder.js

```diff
@@ -67,37 +37,26 @@ function makeRandomColor() {
   return '#' + hexColor
 }
 
-function createFolderComponent({id, removable = true}) {
+function createFolderComponent({removable}) {
+  const ChildFolder = createFolderComponent({removable: true})
+
   function Folder(sources) {
-    function createFolder(childId) {
-      return createFolderComponent(({id: childId}))(sources)
-    }
+    const {addChild$, remove$} = intent(sources.DOM)
 
-    const proxyChildAction$ = xs.create();
-    const action$ = intent(sources.DOM, proxyChildAction$, id)
-    const children$ = model(action$, createFolder)
-    const color = makeRandomColor()
-    const vdom$ = children$.map(makeView(removable, color))
-    const removeSelf$ = action$.filter(({type}) => type === 'removeSelf')
+    const children$ = Collection(ChildFolder, sources, addChild$)
 
-    const childAction$ = children$
-      .map(children => {
-        const childActionStreams = Array.from(children.values())
-          .map(c => c.action$)
-        return xs.merge(...childActionStreams)
-      })
-      .flatten()
+    const color = makeRandomColor()
 
-    proxyChildAction$.imitate(childAction$.endWhen(removeSelf$.take(1)))
+    const vdom$ = Collection.pluck(children$, 'DOM')
+      .map(makeView(removable, color))
 
     return {
       DOM: vdom$,
-      action$,
-      id,
+      remove$
```

Comment:
The signature for collection is now `Collection(component, sources, add$, removalPropName = 'remove$')`.

So `Collection` internally constructs a stream of add/remove reducers and folds over it.

I'm still not sure how I feel about this new approach. It's less boilerplate, but it's also less explicit.